### ISSUE TITLE
eslint-config-react-native-community uses a deprecated version of babel-eslint that can cause eslint to fail when linting template literals

### ISSUE THREAD

**Issue by rchristian** [facebook/react-native#31643]
## Description
The `eslint-config-react-native-community` package currently uses `babel-eslint` as a dependency. This package contains a bug that can cause eslint to fail when encountering template literals. Furthermore, `babel-eslint` has been deprecated and is now `@babel/eslint-parser`, meaning this bug will continue to cause issues until the deprecated package is updated.

See [here](https://github.com/eslint/eslint/issues/12750l) and [here](https://github.com/babel/babel/issues/10904#issuecomment-767275502) for an error report about the bug and confirmation that `babel-eslint` has been deprecated for `@babel/eslint-parser`.

## React Native version:
```
System:
    OS: macOS 11.3.1
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
    Memory: 56.68 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.1.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.12.1 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7199119
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.10 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.64.1 => 0.64.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce
1. Add `eslint` and `eslint-config-react-native-community` as usual to a project containing template literals. Do not use prettier.
2. Add a rule such as 
```
'indent': [2, 4, {
            FunctionExpression: {
                body: 1,
                parameters: 1
            },
            SwitchCase: 1,
        }]
```
to `.eslintrc.js`
3. Try to lint the project

## Expected Results
Lint runs without errors

## Snack, code example, screenshot, or link to a repository:
Deprecation warning:
<img width="807" alt="Screen Shot 2021-06-01 at 10 50 25 AM" src="https://user-images.githubusercontent.com/7800833/120353537-5a0cc980-c2c7-11eb-9e0f-7eefea76ca12.png">

Error when trying to use Yarn to lint a file containing a template literal
<img width="982" alt="Screen Shot 2021-06-01 at 10 39 04 AM" src="https://user-images.githubusercontent.com/7800833/120353740-7a3c8880-c2c7-11eb-9def-f28c65ecef54.png">

A current workaround is to ignore template literals in eslint rules that cause this error. So, for example, adding `ignoredNodes: ['TemplateLiteral']` to the above indent rule's `FunctionExpression` object would allow lint to run successfully.

**Comment by clintonb** [comment]
Related
- https://github.com/facebook/react-native/pull/31182
- https://github.com/facebook/react-native/pull/32431

**Comment by cortinico** [comment]
Fixed by https://github.com/facebook/react-native/pull/34020

I've also just released version 3.0.3 which depends on `@babel/eslint-parser`
